pkg/proxy/iptables/proxier.go

```diff
@@ -1066,6 +1066,29 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 		if svcInfo.ExternalPolicyLocal() {
 			externalPolicyChain = localPolicyChain
+		// Prefer local endpoint for the DNS service.
```

Comment:
This looks like it got cherry-picked into the wrong place in the code. There was a LOT of reorganization of `syncProxyRules()` in 1.24. You'll have to look carefully at where this special case was in 4.10 and then figure out where that code ended up in 1.24.